File: Glyph/Glyph.py
class Glyph:
    def print(self):
        pass

class Line(Glyph):
    def __init__(self, elements):
        self.elements = elements

    def print(self):
        for element in self.elements:
            element.print()

class Char(Glyph):
    def __init__(self, c):
        self.c = c

    def print(self):
        print(self.c)

def main():
    c1 = Char('a')
    c2 = Char('b')
    
    elements = [c1, c2]
    
    l = Line(elements)
    l.print()

if __name__ == "__main__":
    main()

File: Glyph/Glyph.java
public abstract class Glyph {
    public abstract void print();

    public static void main(String[] args) {
        Char c1 = new Char('a');
        Char c2 = new Char('b');
        
        Glyph elements[] = new Glyph[2];
        elements[0] = c1;
        elements[1] = c2;
        
        Line l = new Line(elements);
        l.print();
    }

}
